### PR TITLE
In 150898493 save contact

### PR DESCRIPTION
**Pull request by NateOSI**
### Pivotal Story

- [**INVESTIGATIONS** User can save the data in a contact #150898493](https://www.pivotaltracker.com/story/show/150898493)

### Purpose
**As an** investigator
**I need** to be able to save a record of my work
**So that** the information learned during the contact is not forgotten, and can be referred back to even years later

### Testing Notes
* Until the contacts log stories are played, the only way to return to the same contact created is to copy the URL before saving and then manually paste it back into the browser after saving
* We are not yet attempting to save real data to legacy, until the API is fully complete
* In the short term, this will always display the same set of mock data (see below)
* Be sure to double-check Tony's comment from https://www.pivotaltracker.com/story/show/150411682

### CODE REVIEW

File: app/javascript/investigations/ContactShow.jsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import ShowField from 'common/ShowField'
+import {dateTimeFormatter} from 'utils/dateFormatter'
```
Comment (saraOSI) [inline app/javascript/investigations/ContactShow.jsx]: I would rather have us do the data formatting at the container level. It increases our reusability across service teams, and also makes the view side of the component even dumber. I'm like 60/40 in favor of moving it to the container, so I'm ok leaving it here if you think there's a good reason to do so.

File: app/javascript/reducers/contactReducer.js [hunk 2]
```diff
@@ -1,38 +1,47 @@
...
+      communication_method: {
+        value: null,
+        touched: false,
+      },
+      location: {
+        value: null,
+        touched: false,
+      },
+    })
+    return NEW_CONTACT.setIn(
```
Comment (saraOSI) [inline app/javascript/reducers/contactReducer.js]: Any particular reason for changing how we construct an empty contact?

File: app/javascript/screenings/ScreeningPage.jsx [hunk 3]
```diff
@@ -213,7 +213,7 @@ export class ScreeningPage extends React.Component {
   renderFooterLinks() {
     return (
       <div>
-        <IndexLink to={IntakeConfig.basePath()} className='gap-right'>Home</IndexLink>
+        <IndexLink to={'/'} className='gap-right'>Home</IndexLink>
```
Comment (saraOSI) [inline app/javascript/screenings/ScreeningPage.jsx]: Will this include the proper base path?

File: app/javascript/selectors/contactSelectors.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
+import {createSelector} from 'reselect'
+
+const systemCodeDisplayValue = (code, systemCodes) => systemCodes.find((systemCode) => systemCode.get('code') === code)
```
Comment (saraOSI) [inline app/javascript/selectors/contactSelectors.js]: This appears to be returning the entire code, not just the display value. Maybe there's a better name for this function? `currentSystemCode` or `findSystemCodeByCode` or...?

File: app/javascript/investigations/ContactShow.jsx [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+          </div>
+          <div className='row'>
+            <ShowField gridClassName='col-md-12' label='Location'>
+              {location}
+            </ShowField>
+          </div>
+        </div>
+        <div className='col-md-6'>
+          <div className='row'>
+            <ShowField gridClassName='col-md-12' label='Contact Notes (Optional)'>
```
Comment (saraOSI) [inline app/javascript/investigations/ContactShow.jsx]: Do we need the '(Optional)' flag for the notes field in the show view?

File: app/controllers/contacts_controller.rb [hunk 6]
```diff
@@ -5,5 +5,9 @@
 class ContactsController < ApplicationController
   before_action :authenticate_user, if: :authentication_enabled?
 
-  def new; end
+  def new
+    render :show
```
[the remark this answers is not in the document]
Comment (NateOSI) [reply app/controllers/contacts_controller.rb]: We are currently not using these templates for anything, are javascript is loaded from the layout, I didn't want to create another empty template.

File: app/javascript/investigations/ContactShow.jsx (hunk 5 above)
Comment (NateOSI) [reply app/javascript/investigations/ContactShow.jsx]: it is currently in the Mock

File: app/javascript/reducers/contactReducer.js (hunk 2 above)
Comment (NateOSI) [reply app/javascript/reducers/contactReducer.js]: thought it was clearer to point out what default fields are for a contact and what we are actually setting

File: app/javascript/screenings/ScreeningPage.jsx (hunk 3 above)
Comment (NateOSI) [reply app/javascript/screenings/ScreeningPage.jsx]: No. BasePath is handled by the history object. in the same way we link to /screenings/:id not
/basepath/screenings/:id

File: app/controllers/contacts_controller.rb (hunk 6 above)
[the remark this answers is not in the document]
Comment (saraOSI) [reply app/controllers/contacts_controller.rb]: That makes sense.

File: app/javascript/investigations/ContactShow.jsx (hunk 5 above)
Comment (saraOSI) [reply app/javascript/investigations/ContactShow.jsx]: That's weird, but ok.

File: app/javascript/reducers/contactReducer.js (hunk 2 above)
Comment (saraOSI) [reply app/javascript/reducers/contactReducer.js]: Okie dokie. I'm cool with it either way. Was just curious what your motivation was.

File: app/javascript/screenings/ScreeningPage.jsx (hunk 3 above)
Comment (saraOSI) [reply app/javascript/screenings/ScreeningPage.jsx]: Ok. If this is fixing a bug, then we should call it out somewhere and make sure to test it.

File: app/javascript/investigations/Contact.jsx [hunk 7]
```diff
@@ -37,9 +50,9 @@ class Contact extends React.Component {
                 <div className='row'>
                   <DateField
                     gridClassName='col-md-12'
-                    id='started_at'
+                    id='startedAt'
```
Comment (saraOSI) [inline app/javascript/investigations/Contact.jsx]: Our scss linter is set to use `hyphenated_BEM` . So, while the original ID name is likely wrong, I don't think switching it to camel case is the proper step, either. Here's the docs on (hyphenated BEM)[https://csswizardry.com/2013/01/mindbemding-getting-your-head-round-bem-syntax/]. We probably have a lot of updates to make in the app...

File: spec/javascripts/sagas/createContactSagaSpec.js [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+import {
+  createSuccess,
+  createFailure,
+  CREATE_CONTACT,
+} from 'actions/contactActions'
+
+describe('createContactSaga', () => {
+  it('creates contact on CREATE_CONTACT', () => {
+    const gen = createContactSaga()
+    expect(gen.next().value).toEqual(takeEvery(CREATE_CONTACT, createContact))
```
Comment (brunoOSI) [inline spec/javascripts/sagas/createContactSagaSpec.js]: This isn't readable at all. What is going on here?

File: app/javascript/investigations/ContactShow.jsx (hunk 1 above)
Comment (brunoOSI) [reply app/javascript/investigations/ContactShow.jsx]: +1

File: app/javascript/investigations/ContactShowContainer.jsx [hunk 9]
```diff
@@ -0,0 +1,29 @@
+import ContactShow from 'investigations/ContactShow'
```
Comment (brunoOSI) [inline app/javascript/investigations/ContactShowContainer.jsx]: +1 for adding the container terminology to the codebase

File: config/application.rb [hunk 10]
```diff
@@ -29,7 +29,7 @@ class Application < Rails::Application # :nodoc:
     URL
 
     config.intake = {
-      base_path: ENV.fetch('BASE_PATH', '/'),
+      base_path: ENV.fetch('BASE_PATH', ''),
```
Comment (brunoOSI) [inline config/application.rb]: It will be interesting to see how this plays out with preint.

**Pull request merged**